projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.component.html

```diff
@@ -137,11 +148,14 @@
 </div>
 
 <div class="igx-grid__scroll" [style.height]="'18px'" #scr [hidden]="isHorizontalScrollHidden">
-    <div class="igx-grid__scroll-start" [style.width.px]='pinnedWidth' [hidden]="pinnedWidth === 0"></div>
-    <div class="igx-grid__scroll-main" [style.width.px]='unpinnedWidth'>
+    <div class="igx-grid__scroll-start" 
+        [style.width.px]='pinnedWidth ? getScrollbarSize : (featureColumnsWidth() ? featureColumnsWidth() : 0)'
+        [hidden]="!pinnedWidth ? (!featureColumnsWidth() ? true : false) : false"></div>
+    <div class="igx-grid__scroll-main" [style.width.px]='unpinnedWidth - featureColumnsWidth()'>
         <ng-template igxGridFor [igxGridForOf]='[]' #scrollContainer>
         </ng-template>
     </div>
+    <div class="igx-grid__scroll-end" [style.width.px]='pinnedWidth - featureColumnsWidth()' *ngIf="!isPinningToStart"></div>
```

Comment:
No need to subtract the feature's columns. This breaks the end scrollbar size:
![image](https://user-images.githubusercontent.com/10397980/75141187-1b1dba80-56f9-11ea-9800-3d38d0525139.png)